### PR TITLE
List your issues using 'issue list' and show issue info via 'issue show'

### PR DESCRIPTION
**Pull request by slack-coder**
Add support for listing a users assigned issue via 'issue list', and show issue information with 'issue show <issue id>'

These changes depend on the earlier pull request (https://github.com/radicle-dev/heartwood/pull/159)

### CODE REVIEW

File: radicle-cli/src/commands/issue.rs [hunk 1]
```diff
@@ -232,13 +257,18 @@ pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
...
-        Operation::List => {
+        Operation::List { assignee } => {
             let mut t = term::Table::new(term::table::TableOptions::default());
             for result in issues.all()? {
                 let (id, issue, _) = result?;
+                let assigned: Vec<_> = issue.assigned().collect();
+
+                if Some(true) == assignee.map(|a| !assigned.contains(&&a)) {
+                    continue;
+                }
```
Comment (cloudhead) [inline radicle-cli/src/commands/issue.rs:281]: We don't need to call `issuee.assigned()` if `assignee` is `None`, so I'd rewrite this with that in mind.

File: radicle-cli/src/commands/issue.rs [hunk 2]
```diff
@@ -68,12 +70,14 @@ pub enum Operation {
...
-    List,
+    List {
+        assignee: Option<cob::ActorId>,
+    },
 }
 
 #[derive(Debug)]
 pub struct Options {
-    pub op: Operation,
+    pub op: Option<Operation>,
```
Comment (cloudhead) [inline radicle-cli/src/commands/issue.rs:85]: I find this change complicates a lot of the logic. I would keep it simple and default to showing all issues, which I think is a sensible default (ie. assignee: `None`)

File: radicle-cli/src/commands/issue.rs (hunk 1 above)
Comment (slack-coder) [reply radicle-cli/src/commands/issue.rs:281]: Hmm its used a little further below to be displayed to the users.  Or did I miss something?

The assignee is intended to be whom to list issues for.  The None part of the option is to specify an 'all'.  Its an optional argument.

File: radicle-cli/src/commands/issue.rs [hunk 3]
```diff
@@ -232,13 +255,19 @@ pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
                 )?;
             }
         }
-        Operation::List => {
+        Operation::Default => panic!("default should be set earlier"),
```
Comment (slack-coder) [inline radicle-cli/src/commands/issue.rs:271]: The text needs to be updated, suggestion: "default operation must be set to another"

File: radicle-cli/src/commands/issue.rs [hunk 4]
```diff
@@ -18,11 +20,12 @@ pub const HELP: Help = Help {
     usage: r#"
 Usage
 
+    rad issue
     rad issue new [--title <title>] [--description <text>]
     rad issue state <id> [--closed | --open | --solved]
     rad issue delete <id>
     rad issue react <id> [--emoji <char>]
-    rad issue list
+    rad issue list <assignee>
```
Comment (FintanH) [inline radicle-cli/src/commands/issue.rs:28]:
```suggestion
    rad issue list [<assignee>]
```

Comment (FintanH) [inline radicle-cli/src/commands/issue.rs:28]: `show` is missing from the list

File: radicle-cli/src/commands/issue.rs [hunk 5]
```diff
@@ -171,13 +196,24 @@ pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
     let repo = storage.repository(id)?;
     let mut issues = Issues::open(*signer.public_key(), &repo)?;
 
-    match options.op {
+    let op = match options.op {
+        Operation::Default => Operation::List {
+            assignee: Some(*profile.id()),
+        },
+        _ => options.op,
+    };
```
Comment (FintanH) [inline radicle-cli/src/commands/issue.rs:204]: Why do we need the `Default`? Can't `List` stay as the default operation?

File: radicle-cli/src/commands/issue.rs [hunk 6]
```diff
@@ -171,13 +196,24 @@ pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
...
+
+    match op {
         Operation::Create {
             title: Some(title),
             description: Some(description),
         } => {
             issues.create(title, description, &[], &signer)?;
         }
+        Operation::Show { id } => {
+            let issue = issues.get(&id)?.context("No issue with such ID exists")?;
```
Comment (FintanH) [inline radicle-cli/src/commands/issue.rs:214]: Reads a bit better to me
```suggestion
            let issue = issues.get(&id)?.context("No issue for the given ID exists")?;
```

File: radicle-cli/src/commands/issue.rs [hunk 7]
```diff
@@ -245,3 +299,17 @@ pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
...
+    println!("title: {}", issue.title());
+    println!("state: {}", issue.state());
+
+    let tags: Vec<String> = issue.tags().cloned().map(|t| t.into()).collect();
+    println!("tags: {}", tags.join(", "));
+
+    let assignees: Vec<String> = issue.assigned().map(|a| a.to_string()).collect();
+    println!("assignees: {}", assignees.join(", "));
+
+    println!("{}", issue.description().unwrap_or(""));
```
Comment (FintanH) [inline radicle-cli/src/commands/issue.rs:313]: Should these be `terminal::info`?

File: radicle-cli/src/commands/issue.rs (hunk 5 above)
Comment (slack-coder) [reply radicle-cli/src/commands/issue.rs:204]: The behavior is defined as one of the 'dogfooding cli' issues https://github.com/radicle-dev/heartwood/issues/148

File: radicle-cli/src/commands/issue.rs (hunk 7 above)
Comment (slack-coder) [reply radicle-cli/src/commands/issue.rs]: Updated to use info(..), but not entirely sure when and when it should be preferred.

File: radicle-cli/src/commands/issue.rs (hunk 6 above)
Comment (slack-coder) [reply radicle-cli/src/commands/issue.rs]: :+1:

File: radicle-cli/src/commands/issue.rs (hunk 5 above)
Comment (FintanH) [reply radicle-cli/src/commands/issue.rs:205]: Sorry, what's that got to do with this `Default` thing?

Comment (slack-coder) [reply radicle-cli/src/commands/issue.rs:205]: Maybe I misunderstood the question, the task is to make 'rad issue' list only your assigned issues, which requires your profile ID.

Fetching the profile information happens in run(..), and it didnt feel right fetching it when processing the command's arguments.

Comment (FintanH) [reply radicle-cli/src/commands/issue.rs:205]: Ah, I think I see now. I was thinking that if the `assignee` was `None` you could default to the profile. But I see now that there are three distinct functionalities.

I'd suggest capturing the listing functionality in a function and not doing this pre-match to also avoid the panic below.

Comment (slack-coder) [reply radicle-cli/src/commands/issue.rs]: How do you feel about this approach? https://github.com/radicle-dev/heartwood/pull/171/commits/422e28265f6a3fca1ef120390041d91a8b970891

**Review by FintanH — APPROVED** [review]
Chef's kiss :kissing_closed_eyes: :ok_hand:

**Pull request merged**